static/js/src/advantage/subscribe/react/PurchaseModal.jsx

```diff
@@ -235,102 +100,26 @@ const PurchaseModal = () => {
-                    <Col size="12">
-                      <CheckboxInput
-                        name="TermsCheckbox"
-                        onChange={(e) => {
-                          setTermsChecked(e.target.checked);
-                        }}
-                        label={
-                          <>
-                            I agree to the{" "}
-                            <a
-                              href="/legal/ubuntu-advantage-service-terms"
-                              target="_blank"
-                              rel="noopener noreferrer"
-                            >
-                              Ubuntu Advantage service terms
-                            </a>
-                          </>
-                        }
-                      />
-                    </Col>
-                  </Row>
-                )}
-              </>
-            )}
-          </div>
-          <footer className="p-modal__footer">
-            <Row className="u-no-padding">
-              <ActionButton
-                className="js-cancel-modal col-small-2 col-medium-2 col-start-medium-3 col-start-large-7 col-3 u-no-margin"
-                appearance="neutral"
-                aria-controls="purchase-modal"
-                style={{ textAlign: "center" }}
-              >
-                Cancel
-              </ActionButton>
-
-              {step === 1 ? (
-                <ActionButton
-                  disabled={(!userInfo && !dirty) || !isValid || !isCardValid}
-                  appearance="positive"
-                  className="col-small-2 col-medium-2 col-3 u-no-margin"
-                  style={{ textAlign: "center" }}
-                  onClick={submitForm}
-                  loading={isSubmitting}
-                >
-                  Continue
-                </ActionButton>
-              ) : (
-                <ActionButton
-                  className="col-small-2 col-medium-2 col-3 u-no-margin"
-                  appearance="positive"
-                  style={{ textAlign: "center" }}
-                  disabled={!areTermsChecked}
-                  onClick={onPayClick}
-                  loading={
-                    purchaseMutation.isLoading || isPendingPurchaseLoading
-                  }
-                >
-                  Pay
-                </ActionButton>
-              )}
-            </Row>
-          </footer>
-        </>
-      )}
+      <>
+        {step === 1 ? (
+          <StepOne error={error} closeModal={closeModal} />
+        ) : (
+          <StepTwo
+            setStep={setStep}
+            error={error}
+            setError={setError}
+            closeModal={closeModal}
+          />
+        )}
+        <button
+          ref={buttonRef}
+          aria-controls="purchase-modal"
+          aria-hidden="true"
+          style={{ position: "absolute", left: "-9999px", opacity: 0 }}
+        >
+          Cancel
+        </button>
```

Comment:
This is a workaround to keep and use the event listener set by the js form. (The displaying of the modal is done by the form behind it)

Reply:
Could we display the elements that can trigger the modal via react instead/connect them with the redux store to avoid such workarounds?

Reply:
Yes, but to do that the easiest thing would be to bring all the product selector to the react app. And for the record, I'm planning to do that in a near future*